### PR TITLE
feat: Add `NumericControl`

### PR DESCRIPTION
**Pull request by NathanFarmer**
- Add `InputNumber` renderer
- Add `NumericControl`
- Add `NumericControl` tests
- Add tester
- Add `NumericControl` stories

### CODE REVIEW

File: src/antd/InputNumber.tsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const schema = props.schema
+  const ariaLabel = props.label || schema.description || "Value"
+
+  const defaultValue = schema.default as number | undefined
+  const isDataNonNullObject = typeof props.data === "object" && props.data !== undefined && props.data !== null
+  const isDataEmptyObj = isDataNonNullObject ? Object.keys(props.data as object).length === 0 : false
+  const value = props.data === undefined || isDataEmptyObj ? defaultValue : props.data as number
+
+  const numberType = schema.type
+  const isInteger = (typeof numberType === "string" && numberType === "integer") || (Array.isArray(numberType) && numberType.includes("integer"))
```
Comment (DrewHoo) [inline src/antd/InputNumber.tsx:20]: is there a test case for the Array.isArray(numberType) bit?

**Review by DrewHoo — APPROVED** [review]

File: src/antd/InputNumber.tsx [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        props.handleChange(props.path, coerceToInteger(value))
+      } else {
+        props.handleChange(props.path, coerceToNumber(value))
+      }
+    } else {
+      props.handleChange(props.path, value)
+    }
+  }
+
+  const addonAfter = props.uischema.options?.addonAfter as string | undefined
```
Comment (DrewHoo) [inline src/antd/InputNumber.tsx:33]: is this cast still necessary?

File: src/antd/InputNumber.tsx [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      } else if (numberType === "integer") {
+        return Math.round(parseFloat(value))
+      } else {
+        return parseFloat(value)
+      }
+    }
+    // this allows us to return undefined for cases where the value has been deleted
+    // when InputNumber is paired with a Slider, the Slider value selector will disappear
+    // for required fields an error message will show instead of jumping to some default value
+    return undefined as unknown as number
```
Comment (DrewHoo) [inline src/antd/InputNumber.tsx:67]: this bothers me but antd's type for the parser prop is also weird so I think it's ok (why should it be expected to return a number if value can be undefined?). It'd be a little more palatable if the cast were done higher up instead, e.g. 
```
return <AntdInputNumber
  parser={parser as AntdInputNumber['props']['parser']}
```
idk if consecutive lookups are actually allowed but this is the idea^

File: src/controls/utils.ts [hunk 4]
```diff
@@ -7,3 +7,7 @@ export function decimalToPercentage(value?: number) {
 export function percentageStringToDecimal(value: string | undefined) {
   return Number(value) / 100
 }
+
+export const coerceToInteger = (value: number) => Math.round(value)
+
+export const coerceToNumber = (value: number) => Number(value)
```
Comment (DrewHoo) [inline src/controls/utils.ts:13]: What do you think about deleting these altogether and solving the coercion problem in the anyOf or array controls?

File: src/antd/InputNumber.tsx (hunk 2 above)
Comment (NathanFarmer) [reply src/antd/InputNumber.tsx:33]: Yes, it's considered an `any` until I do this case. It's essentially the same as what we decided to do [here](https://github.com/great-expectations/jsonforms-antd-renderers/blob/bda0b367aa53a0f3ecb0466b7be5a1bfe1f26074/src/controls/AlertControl.tsx#L6).

Comment (NathanFarmer) [reply src/antd/InputNumber.tsx]: I added a new type `InputNumberOptions` to `ui-schema.ts` to precisely follow the pattern established by `AlertControl`. There is still a cast here though, the cast is just on `options` now and it passes through the props types.

File: src/controls/utils.ts (hunk 4 above)
Comment (NathanFarmer) [reply src/controls/utils.ts:13]: It sounds like a better approach. I think we would also need to solve it in `ObjectControl`. Can we defer that for another PR?

File: src/antd/InputNumber.tsx (hunk 2 above)
Comment (DrewHoo) [reply src/antd/InputNumber.tsx]: ah, it's a lot easier to read if you pull off an object and type it:
```
const options: Maybe<InputNumberOptions> = props.uischema.options as InputNumberOptions
const addonAfter = options?.addonAfter
```

Comment (NathanFarmer) [reply src/antd/InputNumber.tsx]: I'm sorry, how is this different than what I did?

Comment (DrewHoo) [reply src/antd/InputNumber.tsx]: typos like ```props.uischema.options?.adonafter as string | undefined``` doesn't get caught but it would in my example

Comment (NathanFarmer) [reply src/antd/InputNumber.tsx]: Ok I don't think you've seen the updated version. This is now outdated.

File: src/antd/InputNumber.tsx (hunk 1 above)
Comment (NathanFarmer) [reply src/antd/InputNumber.tsx]: I added a missing test that asserts that numbers get rounded on input for integer. I added schemas with `type: "integer"` and `type: ["integer"]`. I decided that type arrays with more types than `"integer"` will cause this variable to get set to `false`.

File: src/antd/InputNumber.tsx (hunk 3 above)
Comment (NathanFarmer) [reply src/antd/InputNumber.tsx]: Makes sense to me. I made the change. ✅

**Review by DrewHoo — APPROVED** [review]

**Pull request merged**